sessions/backend-create/products/README.md

```diff
@@ -0,0 +1,115 @@
+
+- the database should be called `fish-shop`,
+- there should be two collections: `products` and `reviews`,
+- download and extract the [resources](README.md#resources) and
+- import the data of the `products.json` file into your `products` collection;
+- import the data of the `reviews.json` file into your `reviews` collection.
+
+Create a `.env.local` file based on the [`.env.local.example`](./.env.local.example). Be sure to spell the name of the database (`fish-shop`) correctly.
+
+### Introduction
+
+Run `npm run dev` and open `localhost:3000` in your browser.
+
+Have a look around:
+
+- there is an overview page with a form to add a new fish and a list of all products below that form;
+- when clicking a product in the list, you'll be redirected to a details page;
+- note that submitting the form does not do anything right now.
+
+Your task is to refactor the app so that submitting the form creates a new entry in your local MongoDB and refreshes the page to show all products (including the new one).
+
+### Add a `POST` route
+
+Switch to [`pages/api/products/index.js`](./pages/api/products/index.js) and write the code for the `request.method` `POST` :
+
+- Use a `try...catch` block.
+- Try to:
+  - Save the product data submitted by the form - accessible in `request.body` - to a variable called `productData`.
+  - Pass the `productData` a `new Product(data_goes_here)` and save it in a variable called `product`.
+  - _Wait_ until the new product was saved in the database with `product.save()`.
+  - Respond with a status `201` and the message `{ status: "Product created." }`.
+- If an error occurs:
+  - Log the error to the console.
+  - Respond with a status `400` and the message `{ error: error.message }`.
+
+Submitting the form will not yet work because the form does not know that you've created a `POST` route it can use.
+
+### Send a `POST` request
+
+Switch to [`components/ProductForm/index.js`](./components/ProductForm/index.js):
+
+- There already is an `onSubmit` function which creates a `productData` object with all relevant data.
+
+Your task is to write a fetch for you newly created `POST` route and send the data to your database.
+
+- Fetch the route `"/api/products"`; `await` the return value and save it in a variable called `response`.
+- As a second argument, pass an object to the `fetch()` method which contains
+  - the `method` set to `POST`,
+  - the `body` set the `JSON.stringify()` of `productData`, and
+  - an `headers` object with `"Content-Type": "application/json"`.
+  - The object should look like the following:
+
+```js
+{
+method: "POST",
+body: JSON.stringify(productData),
+headers: {
+    "Content-Type": "application/json",
+}
+```
+
+Before handling the `response`,
+
+- go to the top of the file and import `useSWR` from `swr`;
+- within the function body of the `ProductForm` (but not inside of the `onSubmit` function), add `const products = useSWR("/api/products");` to get access to `/api/products`.
```

Comment:
In this specific instance, for readability's sake, I'd argue against this. I'd prefer a verbose approach, such as `products.mutate()` instead of `mutate()` to highlight what's happening.